src/engine/renderer/tr_shader.cpp

```diff
@@ -5698,7 +5700,7 @@ static void ValidateStage( shaderStage_t *pStage )
 static float DetermineShaderSort()
 {
 	// fogonly shaders don't have any stage passes
-	if ( numStages == 0 && !shader.isSky )
+	if ( shader.isFog && !shader.isSky )
```

Comment:
I think you could just remove this whole if statement and remove `isFog`. In the place where `isFog` is set, it already puts `SS_FOG`.

Reply:
Also fog shaders are supposed to have `contentparm fog`, so `shader.contentFlags & CONTENTS_FOG`.